x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/filter/SecurityActionFilter.java

```diff
@@ -150,6 +150,7 @@ the system itself (e.g. pings, update mappings, share relocation, etc...) and we
          it to the action without an associated user (not via REST or transport - this is taken care of by
          the {@link Rest} filter and the {@link ServerTransport} filter respectively), it's safe to assume a system user
          here if a request is not associated with any other user.
+         Because we want to fallback to the SystemUser, we don't allow anonymous (AnonymousUser) requests.
```

Comment:
This comment found its way back here : https://github.com/elastic/elasticsearch/pull/52094#discussion_r377350691

Reply:
Splitting this branch up ended up being hard work.
It was worth it because each of the changes was substantial, but there was a lot of rebasing and merging.